### PR TITLE
User interaction logging

### LINKED ISSUE #21: Implement log rolling

### PR DESCRIPTION
**Pull request by edwinzhng**
Fixes #21 

* Added logging for queries, responses, text expansion/collapse, clicks
* A new log file will be created every day with the date appended to the filename as search.log.YYYY-MM-DD

Example logs:
```
{"query_id": "d92fe5f8-229d-4916-8c20-aaffc1e40672", "type": "expanded", "result_id": "t8azymo7", "position": 1, "timestamp": "2020-04-11T03:45:31.593015"}
{"query_id": "d92fe5f8-229d-4916-8c20-aaffc1e40672", "type": "collapsed", "result_id": "t8azymo7", "position": 1, "timestamp": "2020-04-11T03:45:31.593015"}
{"query_id": "d92fe5f8-229d-4916-8c20-aaffc1e40672", "type": "clicked", "result_id ": "mvdq6fw0", "position": 0, "timestamp": "2020-04-11T03:45:31.593015"}
{"query_id": "58cc7336-26d6-474a-8bac-407c69e4fb7d", "type": "query", "vertical": "cord19", "query": "what is the incubation period of covid-19?", "timestamp": "2020-04-11T03:45:31.593015", "response": [...]}
```

### CODE REVIEW

File: api/app/models.py [hunk 1]
```diff
@@ -26,3 +26,22 @@ def validate_highlights(cls, v, values):
...
+    result_id: str
+    position: int
+
+class SearchLogType(str, Enum):
+    query = 'query'
+    collapsed = 'collapsed'
+    expanded = 'expanded'
+    clicked = 'clicked'
+
+class SearchVertical(str, Enum):
```
Comment (rodrigonogueira4) [inline api/app/models.py:45]: Sorry if this is a dummy question, but is vertical a commonly used name? Maybe "SearchCorpus"?

Comment (edwinzhng) [reply api/app/models.py:45]: It's a pretty common term at some companies, I used it at LinkedIn a lot

Comment (edwinzhng) [reply api/app/models.py:45]: For example this post: https://engineering.linkedin.com/search/unifying-linkedin-search-experience

Comment (rodrigonogueira4) [reply api/app/models.py:45]: Cool, thanks! I didn't know about this term

**Pull request merged**